presto-common/src/test/java/com/facebook/presto/common/zorder/TestZOrderIntegers.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static org.testng.Assert.fail;
+
+public class TestZOrderIntegers
+{
+    private static int getHighestSetBitPosition(int value)
+    {
+        // Assumes value is non-negative
+        int position = 0;
+        while (value != 0) {
+            value >>= 1;
+            position++;
+        }
+        return position;
+    }
+
+    @Test
+    public void testZOrderNegativeIntegers()
+    {
+        ZOrder zOrder = new ZOrderIntegers(new ArrayList<>(Arrays.asList(8, 8, 8, 8, 8, 8)));
+        List<Integer> intColumns = new ArrayList<>(Arrays.asList(1, 2, 3, -10, 4, 5));
+
+        try {
+            zOrder.encodeToByteArray(intColumns);
+            fail("Expected test to fail: z-ordering does not support negative integers.");
+        }
+        catch (IllegalArgumentException e) {
+            String expectedMessage = "Our current Z-Ordering implementation does not support negative input numbers.";
+            if (!nullToEmpty(e.getMessage()).equals(expectedMessage)) {
+                fail(format("Expected exception message '%s' to match '%s'", e.getMessage(), expectedMessage));
+            }
+        }
+    }
+
+    @Test
+    public void testZOrderDifferentListSizes()
+    {
+        List<Integer> bitPositions = new ArrayList<>(Arrays.asList(8, 8, 8, 8, 8, 8, 8, 8));
+        ZOrder zOrder = new ZOrderIntegers(bitPositions);
+        List<Integer> intColumns = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));
```

Comment:
Use `ImmutableList.of` once moved to presto-hive-common. Same for all other places. Let's avoid using `ArrayList` whenever possible